### PR TITLE
[PWGHF] taskLc: Remove track loop in the fill histogram function of data

### PR DESCRIPTION
**Pull request by zhangbiao-phy**
This pr maybe can fix the issue of CPU consumption when run the task using real data

### CODE REVIEW

**Review by mfaggin — APPROVED** [review]

File: PWGHF/D2H/Tasks/taskLc.cxx [hunk 1]
```diff
@@ -547,25 +547,14 @@ struct HfTaskLc {
...
-        }
-        if (std::abs(track.dcaXY()) > 0.0025 || std::abs(track.dcaZ()) > 0.0025) {
-          continue;
-        }
-        nTracks++;
-      }
-    }
-    registry.fill(HIST("Data/hMultiplicity"), nTracks);
+    auto numPvContributors = collision.numContrib();
+    registry.fill(HIST("Data/hMultiplicity"), numPvContributors);
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskLc.cxx:557]: Number of PV contributors and multiplicity are two different things. Replace the histogram with a correct one if you don't want to store multiplicity anymore.

File: PWGHF/D2H/Tasks/taskLc.cxx [hunk 2]
```diff
@@ -586,12 +575,12 @@ struct HfTaskLc {
 
       if (candidate.isSelLcToPKPi() >= selectionFlagLc) {
         registry.fill(HIST("Data/hMass"), hfHelper.invMassLcToPKPi(candidate));
-        registry.fill(HIST("Data/hMassVsPtVsMult"), hfHelper.invMassLcToPKPi(candidate), pt, nTracks);
+        registry.fill(HIST("Data/hMassVsPtVsMult"), hfHelper.invMassLcToPKPi(candidate), pt, numPvContributors);
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskLc.cxx:578]: Same comment

File: PWGHF/D2H/Tasks/taskLc.cxx [hunk 3]
```diff
@@ -586,12 +575,12 @@ struct HfTaskLc {
...
         registry.fill(HIST("Data/hMassVsPt"), hfHelper.invMassLcToPKPi(candidate), pt);
       }
       if (candidate.isSelLcToPiKP() >= selectionFlagLc) {
         registry.fill(HIST("Data/hMass"), hfHelper.invMassLcToPiKP(candidate));
-        registry.fill(HIST("Data/hMassVsPtVsMult"), hfHelper.invMassLcToPiKP(candidate), pt, nTracks);
+        registry.fill(HIST("Data/hMassVsPtVsMult"), hfHelper.invMassLcToPiKP(candidate), pt, numPvContributors);
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskLc.cxx:583]: Same comment

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by zhangbiao-phy** [comment]
hi @vkucera, Could you please have a look the commit based on you comment? if no, could you please approve it? Thank you so much!

File: PWGHF/D2H/Tasks/taskLc.cxx (hunk 1 above)
Comment (vkucera) [reply PWGHF/D2H/Tasks/taskLc.cxx]: Thanks

**Review by vkucera — APPROVED** [review]

**Review by fcatalan92 — APPROVED** [review]

**Pull request merged**